maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultArtifactDescriptorReader.java

```diff
@@ -275,6 +275,7 @@ private Model loadPom( RepositorySystemSession session, ArtifactDescriptorReques
                 modelRequest.setTwoPhaseBuilding( false );
                 modelRequest.setSystemProperties( toProperties( session.getUserProperties(),
                                                                 session.getSystemProperties() ) );
+                modelRequest.setUserProperties( toProperties( session.getUserProperties(), null ) );
```

Comment:
This also means, that the line above it should be reduced to `modelRequest.setSystemProperties( toProperties( session.getSystemProperties() ) );`
Next you need to analyze where it is used, and figure out if that code should be changed. It is an internal package, so it suggests such changes should be safe.